### PR TITLE
Feature/gdpr

### PR DESCRIPTION
**Pull request by iddanh**

### CODE REVIEW

File: src/user-cookie-provider.js [hunk 1]
```diff
@@ -2,89 +2,123 @@ var xDomainCookie = require('./xdomain_cookie');
 
 (function (exports) {
     "use strict";
+
     function pbUserCookieProvider(guidGenerator, playbuzzHomeUrl, debug) {
```
Comment (pb-nati) [inline src/user-cookie-provider.js]: - [ ] guidGenerator is useless

File: src/user-cookie-provider.js [hunk 2]
```diff
@@ -2,89 +2,123 @@ var xDomainCookie = require('./xdomain_cookie');
...
+            getAnonymous: getAnonymous
+        };
+
+        function init(onFailure) {
+
+            // Try getting the authenticated cookie from the xdomain provider.
+            xdomainCookieProvider.get(AUTHENTICATED_COOKIE_NAME, function (result) {
+
+                // If cookie found, set it.
+                if (result !== null) {
```
Comment (pb-nati) [inline src/user-cookie-provider.js]: - [ ] wrong. always set it. even if it's null.

File: src/user-cookie-provider.js [hunk 3]
```diff
@@ -2,89 +2,123 @@ var xDomainCookie = require('./xdomain_cookie');
...
+        var IFRAME_PATH = playbuzzHomeUrl,
+            IFRAME_POSTMESSAGE_NAMESPACE = '',
+            IFRAME_DEBUG = debug,
+            AUTHENTICATED_COOKIE_NAME = 'playbuzz',
+            ANONYMOUS_COOKIE_NAME = 'pb-tracking-id',
+            COOKIE_EXPIRY_DAYS = 30,
+            CHECK_INTERVAL = 200;
+
+        var xdomainCookieProvider = window.xDomainCookie(IFRAME_PATH, IFRAME_POSTMESSAGE_NAMESPACE, IFRAME_DEBUG),
+            authenticatedCookie = null,
```
Comment (pb-nati) [inline src/user-cookie-provider.js]: - [ ] they should be `undefined` at the start

File: src/user-cookie-provider.js [hunk 4]
```diff
@@ -2,89 +2,123 @@ var xDomainCookie = require('./xdomain_cookie');
...
+        /**
+         * Function that gets the authenticated cookie in an async way.
+         * If a null value is found, it will keep trying to fetch the cookie.
+         * Will not call the given callback until a value was found.
+         * @param callback
+         */
+        function getAuthenticated(callback) {
+
+            // If a user exist, return it.
+            if (authenticatedCookie !== null) {
```
Comment (pb-nati) [inline src/user-cookie-provider.js]: should be `authenticatedCookie !== undefined`

File: src/user-cookie-provider.js [hunk 5]
```diff
@@ -2,89 +2,123 @@ var xDomainCookie = require('./xdomain_cookie');
...
-    // Implementation
+            // If a user does not yet exist, wait for it.
+            var interval = setInterval(function () {
 
-    return {
-        init: init,
-        get: get
-    };
+                // If a user exist, return it, and cancel interval.
+                if (authenticatedCookie !== null) {
```
Comment (pb-nati) [inline src/user-cookie-provider.js]: should be `authenticatedCookie !== undefined`

File: src/user-cookie-provider.js [hunk 6]
```diff
@@ -2,89 +2,123 @@ var xDomainCookie = require('./xdomain_cookie');
...
+        /**
+         * Function that gets the anonymous cookie in an async way. Same as getAuthenticated
+         */
+        function getAnonymous(callback) {
 
-            // If cookie found, set it.
-            if (result !== null) {
-                user = result;
+            // If a id exist, return it.
+            if (anonymousCookie !== null) {
```
Comment (pb-nati) [inline src/user-cookie-provider.js]: should be `anonymousCookie !== undefined`

File: src/user-cookie-provider.js [hunk 7]
```diff
@@ -2,89 +2,123 @@ var xDomainCookie = require('./xdomain_cookie');
...
+            // If a id does not yet exist, wait for it.
+            var interval = setInterval(function () {
 
-        // Subscribe to a case of a failure.
-        xdomainCookieProvider.subscribeToFailure(function (cookieFailed, localStorageFailed) {
-            onFailure(JSON.parse(user), cookieFailed, localStorageFailed);
-        });
-    }
+                // If a id exist, return it, and cancel interval.
+                if (anonymousCookie !== null) {
```
Comment (pb-nati) [inline src/user-cookie-provider.js]: should be `anonymousCookie !== undefined`

File: dist/user-cookie-provider.js [hunk 8]
```diff
@@ -65,298 +65,149 @@
 /************************************************************************/
 /******/ ([
 /* 0 */
-/***/ (function(module, exports, __webpack_require__) {
```
Comment (pb-nati) [inline dist/user-cookie-provider.js]: - [ ] @iddanh hello... you removed this file...

Comment (iddanh) [reply dist/user-cookie-provider.js]: @pb-nati The file was removed because I removed the require line?

**Pull request merged**